pkg/util/timeutil/timeutil.go

```diff
@@ -50,22 +52,47 @@ func DurationString(duration time.Duration) string {
 			// convert to mins
 			durStr = fmt.Sprintf("%dm", durSecs/SecsPerMin)
 		} else if durSecs > 0 {
-			// default to mins, as long as duration is positive
+			// default to secs, as long as duration is positive
 			durStr = fmt.Sprintf("%ds", durSecs)
 		}
 	}
 
 	return durStr
 }
 
+// DurationToPromString returns a Prometheus formatted string with leading offset or empty string if given a negative duration
+func DurationToPromString(duration time.Duration) string {
+	dirStr := DurationString(duration)
+	if dirStr != "" {
+		dirStr = fmt.Sprintf("offset %s", dirStr)
+	}
+	return dirStr
+}
```

Comment:
If I give this function `duration = 30 * time.Minute`, then I get back the string `"offset 30m"`? FWIW the name `DurationToPromString` is misleading here, as not all Prometheus durations need to be prepended with "offset", which also makes it confusing to read the code where this function is used.

Reply:
While I'm open to being convinced otherwise, I tend to think we don't need an entire separate function for prepending "offset" on a duration string.

Reply:
I changed the name to DurationToPromOffsetString which I hope is a bit more clear how it is different from the other method. The function does a little more than just prepend, it also checks for empty strings. I agree that the fact that it through you off is a problem, and if you think even with the new name it isn't clear enough what it does, I will remove it. I do still like it better as a utility function.